include/cinder/AxisAlignedBox.h

```diff
@@ -28,37 +28,178 @@
 
 namespace cinder {
 
-class AxisAlignedBox3f {
- public:
-	AxisAlignedBox3f() {}
-	AxisAlignedBox3f( const vec3 &aMin, const vec3 &aMax );
-
-	bool	intersects( const Ray &ray );
-	int		intersect( const Ray &ray, float intersections[2] );
-
-	vec3			getCenter() const { return ( mExtents[1] + mExtents[0] ) * 0.5f; }
-	vec3			getSize() const { return mExtents[1] - mExtents[0]; }
-
-	//! Expands the box so that it contains \a point
-	void			include( const vec3 &point );
-	//! Expands the box so that it contains \a box
-	void			include( const AxisAlignedBox3f &box );
-	
-	const vec3&	getMin() const { return mExtents[0]; }
-	const vec3&	getMax() const { return mExtents[1]; }
-
-	//! for use in frustum culling
-	vec3	getNegative( const vec3 &normal ) const;
-	vec3	getPositive( const vec3 &normal ) const;
-
-	//! converts axis-aligned box to another coordinate space
-	AxisAlignedBox3f transformed( const mat4 &transform ) const;
-	
-	static bool calcTriangleIntersection( const Ray &ray, const vec3 &vert0, const vec3 &vert1, const vec3 &vert2, float *result );
-
- protected:
-	vec3 mExtents[2];
-	vec3 mVerts[8];
+class AxisAlignedBox {
+public:
+	AxisAlignedBox() : mCenter( 0 ), mExtents( 0 ) {}
+	AxisAlignedBox( const vec3 &min, const vec3 &max ) { set( min, max ); }
+
+	//! Returns the center of the axis-aligned box.
+	const vec3& getCenter() const { return mCenter; }
+
+	//! Returns the extents of the axis-aligned box.
+	const vec3& getExtents() const { return mExtents; }
+
+	//! Returns the size of the axis-aligned box. 
+	vec3 getSize() const { return 2.0f * mExtents; }
+
+	//! Returns the corner of the axis-aligned box with the smallest x, y and z coordinates.
+	vec3 getMin() const { return mCenter - mExtents; }
+
+	//! Returns the corner of the axis-aligned box with the largest x, y and z coordinates.
+	vec3 getMax() const { return mCenter + mExtents; }
+
+	//! Construct an axis-aligned box by specifying two opposite corners.
+	void set( const vec3 &min, const vec3 &max )
+	{
+		mCenter = 0.5f * ( min + max );
+		mExtents = glm::abs( max - mCenter );
+	}
+
+	//! Returns \c true if the axis-aligned box contains \a point.
+	bool contains( const vec3 &point ) const
+	{
+		return glm::all( glm::lessThanEqual( glm::abs( point - mCenter ), mExtents ) );
+	}
+
+	//! Returns \c true if the axis-aligned boxes intersect.
+	bool intersects( const AxisAlignedBox &box ) const
+	{
+		return glm::all( glm::lessThanEqual( glm::abs( box.mCenter - mCenter ), box.mExtents + mExtents ) );
+	}
+
+	//! Returns \c true if the ray intersects the axis-aligned box.
+	bool intersects( const Ray &ray ) const
+	{
+		vec3 min = ( getMin() - ray.getOrigin() ) / ray.getDirection();
```

Comment:
No, it's fine, but thanks for pointing it out. The #INF's will be ignored by the min() and max() functions.